rc/tools/clipboard.kak

```diff
@@ -0,0 +1,94 @@
+declare-option -docstring %{
+    The shell command to which data will be piped to copy to the system clipboard
+} str clipboardcmd_copy %sh{
+    for cmd in "xsel -i" "pbcopy" "wl-copy -p"; do
+        readonly program="${cmd%% *}"
+        if command -v "${program}" >/dev/null 2>&1; then
+            printf %s "${cmd}"
+            exit
+        fi
+    done
+}
+
+declare-option -docstring %{
+    The shell command which will paste the system clipboard into the current buffer
+} str clipboardcmd_paste %sh{
+    for cmd in "xsel -o 2>/dev/null" "pbpaste" "wl-paste -p"; do
+        readonly program="${cmd%% *}"
+        if command -v "${program}" >/dev/null 2>&1; then
+            printf %s "${cmd}"
+            exit
+        fi
+    done
+}
+
+define-command -params .. -docstring %{
+    clipboard-copy [<data>…]: copy the given data to the system clipboard
+
+    If no data is passed as argument to the command, the current selection is used
+    If multiple selections/data arguments are to be stored into the clipboard, they are joined with a newline character
+} clipboard-copy %{ evaluate-commands %sh{
+    join() {
+        # NOTE: this is a POSIX way of storing a newline
+        # character into $IFS without spawning a subshell
+        IFS='
+'
+        printf %s "$*"
+    }
+
+    if [ $# -eq 0 ]; then
+        eval "set -- $kak_quoted_selections"
+    fi
+
+    if ! join "$@" | eval "${kak_opt_clipboardcmd_copy}"; then
+        fail "Unable to copy data to the system clipboard"
+    fi
+} }
+
+define-command -docstring %{
+    Paste the contents of the system clipboard before the selection
+} clipboard-paste-before %{
+    execute-keys ! %opt{clipboardcmd_paste} <ret>
```

Comment:
Add `<space>` here to prevent the paste command being added to the history register?